internal/services/dispatcher/server.go

```diff
@@ -706,6 +706,27 @@ func (s *DispatcherImpl) Unsubscribe(ctx context.Context, request *contracts.Wor
 	}, nil
 }
 
+func (d *DispatcherImpl) RefreshTimeout(ctx context.Context, request *contracts.RefreshTimeoutRequest) (*contracts.RefreshTimeoutResponse, error) {
+	tenant := ctx.Value("tenant").(*dbsqlc.Tenant)
+	tenantId := sqlchelpers.UUIDToStr(tenant.ID)
+
+	stepRun, err := d.repo.StepRun().RefreshTimeoutAt(ctx, tenantId, request.StepRunId, request.IncrementTimeoutBy)
```

Comment:
@abelanger5 is there a convenient way to validate the IncrementTimeoutBy string?

Reply:
Use the `duration` validator within `RefreshTimeoutAt` in an `opts` struct.